src/requests/adapters.py

```diff
@@ -404,7 +414,10 @@ def _get_connection(self, request, verify, proxies=None, cert=None):
         return conn
 
     def get_connection(self, url, proxies=None):
-        """Returns a urllib3 connection for the given URL. This should not be
+        """DEPRECATED: Users should move to `get_connection_with_tls_context`
```

Comment:
If this method is deprecated shall we start emitting `DeprecationWarning`?

Reply:
I thought about that but it's not reachable in any of our code anymore and if someone is using a custom implementation it won't have the warning.

Happy to be wrong if I'm missing something but it seems like it will just be dead code on arrival.